TailLight/device.cpp

```diff
@@ -71,6 +95,35 @@ UNICODE_STRING GetTargetPropertyString(WDFIOTARGET target, DEVICE_REGISTRY_PROPE
 }
 
 
+VOID EvtCleanupCallback(
+    _In_ WDFOBJECT object
+)
+/*++
+Routine Description:
+    Removes the plug and play notification callback if registered.
+    Called near the end of processing an IRP_MN_REMOVE_DEVICE.
+    This work could also be done at EvtDeviceSelfManagedIoCleanup, which
+    comes before this callback. However using the lifetime of the device
+    context means there is little chance of the device context disappearing
```

Comment:
What is meant by "little chance of the device context disappearing from underneath" here?

Doesn't the framework guarantee that the context will outlive the `EvtCleanupCallback` callback? If not, then where is this WDF limitation documented?

I don't think we should write brittle code that is based on finger crossing and hoping for a good outcome. We should instead utilize the design guarantees offered by WDF & WDM.

Reply:
This holds as long as WDF & WDM are bug free. I've seen workarounds or fixes implemented for code that's been around for decades.

Reply:
Please then either assume that WDF & WDM are bug-free, or add comments links to reported WDF/WDM issues.

Reply:
Thanks for removing unreferenced claims from your PR.